include/zephyr/drivers/stepper_motor_controller.h

```diff
@@ -0,0 +1,229 @@
+
+/**
+ * @brief Motor Driver Info Channels
+ * @details This is a collection of general functionalities that a motor driver could offer
+ */
+enum motor_driver_info_channel {
+    /** Freewheeling mode where a motor should run with a certain velocity */
+	FREE_WHEELING,
+    /** Set or Get the actual position from motor controller */
+	ACTUAL_POSITION,
+    /** Set or Get the  target position from motor controller */
+	TARGET_POSITION,
+    /** Set or Get the actual velocity from motor controller */
+	ACTUAL_VELOCITY,
+    /** Set or Get the target velocity from motor controller*/
+	TARGET_VELOCITY,
+    /** Load detection, End Stop status and so on*/
+	STALL_DETECTION,
+    /** Configure stall-guard or threshold allowing controller to detect stalling */
+	STALL_GUARD,
+};
+
+struct stepper_motor_dt_spec {
+    /** bus refers to the device offering motor bus */
+	const struct device *bus;
+    /** addr refers to the index of the motor */
+	int32_t addr;
+};
+
+
+/**
+ * @typedef stepper_motor_controller_reset_t
+ * @brief reset the stepper motor controller to default settings
+ *
+ * @see stepper_motor_controller_reset for more details
+ */
+typedef int32_t (*stepper_motor_controller_reset_t)(const struct stepper_motor_dt_spec *bus);
+
+/**
+ * @typedef stepper_motor_controller_write_t
+ * @brief write to the stepper motor controller using a generalized channel
+ *
+ * @see stepper_motor_controller_write for more details
+ */
+typedef int32_t (*stepper_motor_controller_write_t)(const struct stepper_motor_dt_spec *bus,
+						    int32_t motor_channel, int32_t data);
+
+/**
+ * @typedef stepper_motor_controller_read_t
+ * @brief read a certain value from stepper motor controller via a generalized channel
+ *
+ * @see stepper_motor_controller_read for more details
+ */
+typedef int32_t (*stepper_motor_controller_read_t)(const struct stepper_motor_dt_spec *bus,
+						   int32_t motor_channel, int32_t *data);
+
+/**
+ * @typedef stepper_motor_controller_reg_write_t
+ * @brief write to a certain register in motor controller directly
+ *
+ * @see stepper_motor_controller_write_reg for more details
+ */
+typedef int32_t (*stepper_motor_controller_reg_write_t)(const struct device *dev,
+							uint8_t reg_address, uint32_t reg_value);
+
+/**
+ * @typedef stepper_motor_controller_reg_read_t
+ * @brief read a certain register value from motor controller directly
+ *
+ * @see stepper_motor_read_reg for more details
+ */
+typedef int32_t (*stepper_motor_controller_reg_read_t)(const struct device *dev,
+						       uint8_t reg_address, uint32_t *reg_value);
+__subsystem struct stepper_motor_controller_api {
+	stepper_motor_controller_reset_t stepper_motor_controller_reset;
+	stepper_motor_controller_write_t stepper_motor_controller_write;
+	stepper_motor_controller_read_t stepper_motor_controller_read;
+
+	stepper_motor_controller_reg_write_t stepper_motor_controller_write_reg;
+	stepper_motor_controller_reg_read_t stepper_motor_controller_read_reg;
+};
```

Comment:
this really looks like a kind of MFD device, where you have a generic read/write for multiple elements of the same chip. I think this modeling of a bus is really an overkill. The API could just be multichannel, or, just have a parent MFD and as child nodes as motors the chip has.